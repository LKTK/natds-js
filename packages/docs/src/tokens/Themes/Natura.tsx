import * as React from 'react';
import { themes } from '@naturacosmeticos/natds-styles';
import Table from '@addons/Table/Table';

const { natura } = themes;

function FlattenTheme(accum: any, [key, value]: any): any {
  const type = typeof value;

  if (type === 'function' || type === 'object' && !!value) {
    return Object.entries(value).reduce(FlattenTheme, [key, ...accum]);
  }

  const [parent] = accum;

  if (!parent) return accum;

  const name = `${parent}-${key}`;

  return [...accum, [{ name }, { name: value, type: 'color' }]];
}

function removeStrings(entry: any) {
  return typeof entry !== 'string';
}

const { light, dark } = natura;

const darkSchema = {
  head: ['name', 'color'],
  body: (dark && dark.palette) && Object.entries(dark.palette).reduce(FlattenTheme, []).filter(removeStrings)
};

darkSchema.body.push(
  [
    {
      name: 'webFontFamilyBranding',
      type: 'text'
    },
    {
      name: dark.typography.web.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'androidFontFamilyBranding',
      type: 'text'
    },
    {
      name: dark.typography.android.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'iosFontFamilyBranding',
      type: 'text'
    },
    {
      name: dark.typography.iOS.fontFamilyBrand,
      type: 'text'
    }
  ]
);

const lightSchema = {
  head: ['name', 'color'],
  body: (light && light.palette) && Object.entries(light.palette).reduce(FlattenTheme, []).filter(removeStrings)
};

lightSchema.body.push(
  [
    {
      name: 'webFontFamilyBranding',
      type: 'text'
    },
    {
      name: light.typography.web.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'androidFontFamilyBranding',
      type: 'text'
    },
    {
      name: light.typography.android.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'iosFontFamilyBranding',
      type: 'text'
    },
    {
      name: light.typography.iOS.fontFamilyBrand,
      type: 'text'
    }
  ]
);

export const NaturaLightComponent = () => <Table schema={lightSchema} title="Light" />;
export const NaturaDarkComponent = () => <Table schema={darkSchema} title="Dark" />;
