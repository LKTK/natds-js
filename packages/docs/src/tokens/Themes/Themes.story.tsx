import * as React from 'react';
import { themes } from '@naturacosmeticos/natds-styles';
import Table from '@addons/Table/Table';
import tokensParams from '../shared';

const { natura } = themes;

function FlattenTheme(accum: any, [key, value]: any): any {
  const type = typeof value;

  if (type === 'function' || type === 'object' && !!value) {
    return Object.entries(value).reduce(FlattenTheme, [key, ...accum]);
  }

  const [parent] = accum;

  if (!parent) return accum;

  const name = `${parent}-${key}`;

  return [...accum, [{ name }, { name: value, type: 'color' }]];
}

function removeStrings(entry: any) {
  return typeof entry !== 'string';
}

const { light, dark } = natura;

const darkSchema = {
  head: ['name', 'color'],
  body: (dark && dark.palette) && Object.entries(dark.palette).reduce(FlattenTheme, []).filter(removeStrings)
};

const lightSchema = {
  head: ['name', 'color'],
  body: (light && light.palette) && Object.entries(light.palette).reduce(FlattenTheme, []).filter(removeStrings)
};

const LightComponent = () => <Table schema={lightSchema} title="Light" />;
const DarkComponent = () => <Table schema={darkSchema} title="Dark" />;

export default {
  title: 'Design Tokens|Themes',
  parameters: {
    docs: {
      container: null,
      page: () => (
        <div>
          <Table schema={lightSchema} title="Light" />
          <Table schema={darkSchema} title="Dark" />
        </div>
      )
    },
    ...tokensParams
  }
};

export const NaturaLight = () => (
  <LightComponent />
);

export const NaturaDark = () => (
  <DarkComponent />
);
