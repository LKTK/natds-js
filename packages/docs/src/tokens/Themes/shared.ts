function flattenTheme(accum: any, [key, value]: any): any {
  const type = typeof value;

  if (type === 'function' || (type === 'object' && !!value)) {
    return Object.entries(value).reduce(flattenTheme, [key, ...accum]);
  }

  const [parent] = accum;

  if (!parent) return accum;

  const name = `${parent}-${key}`;

  return [...accum, [{ name }, { name: value, type: 'color' }]];
}

function removeStrings(entry: any) {
  return typeof entry !== 'string';
}

export function parseTheme(theme: any) {
  const schema = {
    head: ['name', 'color'],
    body:
      theme &&
      theme.palette &&
      Object.entries(theme.palette)
        .reduce(flattenTheme, [])
        .filter(removeStrings)
  };

  schema.body.push(
    [
      {
        name: 'fontFamilyBranding1',
        type: 'text',
      },
      {
        name: theme.typography.fontFamilyBrand1
          ? theme.typography.fontFamilyBrand1
          : 'Undefined',
        type: 'text',
      },
    ],
    [
      {
        name: 'fontFamilyBranding2',
        type: 'text',
      },
      {
        name: theme.typography.fontFamilyBrand2
          ? theme.typography.fontFamilyBrand2
          : 'Undefined',
        type: 'text',
      },
    ],
    [
      {
        name: 'fontFamilyBranding3',
        type: 'text',
      },
      {
        name: theme.typography.fontFamilyBrand3
          ? theme.typography.fontFamilyBrand3
          : 'Undefined',
        type: 'text',
      },
    ],
  );

  return schema;
}
