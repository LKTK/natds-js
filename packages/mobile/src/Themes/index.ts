export { themes } from '@naturacosmeticos/natds-styles';
