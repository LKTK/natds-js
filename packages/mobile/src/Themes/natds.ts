import { themes } from '@naturacosmeticos/natds-styles';
import { DefaultTheme, ThemeShape } from 'react-native-paper';

const { natds } = themes;

const natdsTheme: ThemeShape =  {
  ...DefaultTheme,
  ...natds
};

export default natdsTheme;
