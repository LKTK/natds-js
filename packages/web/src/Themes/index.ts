import {
  themes as styleThemes,
} from "@naturacosmeticos/natds-styles";
import { parseTheme } from "./parseTheme";
import { ThemeBrandName } from "./ThemeBrandName";
import createThemesObject from "./createThemesObject";
import { IThemesWeb } from "./IThemesWeb";

export { IThemeWeb } from "./IThemeWeb";

export const themes = Object.keys(styleThemes).reduce((value: IThemesWeb<ThemeBrandName>, key: string) => {

  const result = value;

  result[key as ThemeBrandName].dark = parseTheme(styleThemes[key as ThemeBrandName].dark);
  result[key as ThemeBrandName].light = parseTheme(styleThemes[key as ThemeBrandName].light);

  return result;
}, createThemesObject());
