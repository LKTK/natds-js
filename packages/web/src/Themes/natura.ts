import { themes, tokens } from '@naturacosmeticos/natds-styles';
import { createMuiTheme } from '@material-ui/core';

const { spacing }  = tokens.spacing;
const { naturaLight } = themes;
const { colors, roundness } = naturaLight;

const natdsTheme = createMuiTheme({
  spacing,
  palette: {
    primary: {
      main: colors.primary
    },
    secondary: {
      main: colors.secondary
    },
    error: {
      main: colors.error
    },
    type: 'light',
  },
  shape: {
    borderRadius: roundness
  }
});

export default natdsTheme;
