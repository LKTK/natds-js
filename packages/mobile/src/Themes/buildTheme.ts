import { IThemeShape } from '../Provider/IThemeShape';
import { DefaultTheme } from 'react-native-paper';
import { IThemeMobile } from 'Themes';

export function buildTheme(theme: IThemeMobile | undefined, defaultTheme: IThemeMobile): IThemeShape {
  const provider = theme ? theme : defaultTheme;
  const { palette = {}, shape } = provider;

  const {
    primary: { main: primary } = { main: undefined },
    secondary: { main: secondary } = { main: undefined },
    error: { main: error } = { main: undefined },
    background = {},
    text = {}
  } = palette;

  const defaultColors = DefaultTheme.colors;

  const colors = {
    primary: primary ? primary : defaultColors.primary,
    accent: secondary ? secondary : defaultColors.accent,
    secondary,
    error: error ? error : defaultColors.error,
    background: background.default ? background.default : defaultColors.backdrop,
    surface: background.paper ? background.paper : defaultColors.surface,
    text: text.primary ? text.primary : defaultColors.text,
    onBackground: text.primary ? text.primary : defaultColors.onBackground,
    onSurface: text.secondary ? text.secondary : defaultColors.onSurface,
    disabled: text.disabled ? text.disabled : defaultColors.disabled,
    placeholder: text.hint ? text.hint : defaultColors.placeholder
  };

  const roundness = shape && shape.borderRadius
    ? shape.borderRadius
    : DefaultTheme.roundness;

  const isDarkMode = palette.type === 'dark';

  const newTheme: IThemeShape = {
    ...DefaultTheme,
    dark: isDarkMode,
    roundness,
    colors: {
      ...DefaultTheme.colors,
      ...colors
    }
  };

  return newTheme;
}
