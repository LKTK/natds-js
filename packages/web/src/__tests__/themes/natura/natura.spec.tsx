import { themes } from '../../../Themes';
import naturaLight from './naturaLightThemeParsed.json';
import naturaDark from './naturaDarkThemeParsed.json';

describe('Themes', () => {
  test('It should match the natura light theme snapshot', () => {
    const { natura: { light } } = themes;

    expect(light).toStrictEqual(naturaLight);
  });

  test('it should match the natura dark theme snapshot', () => {
    const { natura: { dark } } = themes;

    expect(dark).toStrictEqual(naturaDark);
  });
});
