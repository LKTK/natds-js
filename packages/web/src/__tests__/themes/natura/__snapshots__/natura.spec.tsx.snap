// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themes It should match the natura light theme snapshot: Natura Light Theme Snapshot 1`] = `
Object {
  "avatarSizes": Object {
    "huge": Object {
      "fontSize": "1.5rem",
      "size": 80,
    },
    "large": Object {
      "fontSize": "1rem",
      "size": 48,
    },
    "small": Object {
      "fontSize": ".875rem",
      "size": 32,
    },
    "standard": Object {
      "fontSize": "1rem",
      "size": 40,
    },
    "tiny": Object {
      "fontSize": ".625rem",
      "size": 24,
    },
  },
  "buttonSizes": Object {
    "large": Object {
      "height": 48,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
    },
    "medium": Object {
      "height": 40,
      "paddingBottom": 0,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 0,
    },
    "small": Object {
      "height": 32,
      "paddingBottom": 0,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
    },
  },
  "iconSizes": Object {
    "huge": 144,
    "hugex": 192,
    "hugexx": 256,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "medium": 64,
    "micro": 16,
    "small": 32,
    "standard": 48,
    "tiny": 24,
  },
  "overrides": Object {
    "MuiAvatar": Object {
      "colorDefault": Object {
        "color": "#333333",
      },
    },
    "MuiBadge": Object {
      "badge": Object {
        "padding": "0 4px",
      },
    },
    "MuiButton": Object {
      "contained": Object {
        "&.Mui-disabled": Object {
          "backgroundColor": "#00000014",
          "color": "#0000003D",
        },
      },
      "outlined": Object {
        "&.Mui-disabled": Object {
          "borderColor": "#0000003D",
          "color": "#0000003D",
        },
        "borderColor": "#0000003D",
        "padding": null,
      },
      "outlinedPrimary": Object {
        "color": "#333333",
      },
      "outlinedSecondary": Object {
        "&.Mui-disabled": Object {
          "borderColor": "#0000003D",
        },
        "color": "#333333",
      },
      "root": Object {
        "fontSize": "0.875rem",
        "fontWeight": 500,
        "height": 40,
        "paddingBottom": 0,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 0,
      },
      "sizeLarge": Object {
        "fontSize": "0.875rem",
        "fontWeight": 500,
        "height": 48,
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
      },
      "sizeSmall": Object {
        "fontSize": "0.875rem",
        "fontWeight": 500,
        "height": 32,
        "paddingBottom": 0,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 0,
      },
      "text": Object {
        "&.Mui-disabled": Object {
          "borderColor": "#0000003D",
          "color": "#0000003D",
        },
        "padding": null,
      },
    },
    "MuiButtonGroup": Object {
      "groupedContainedHorizontal": Object {
        "&:not(:last-child)": Object {
          "&.Mui-disabled": Object {
            "borderColor": "#0000003D",
          },
          "borderRightColor": "#0000001F",
        },
      },
      "groupedContainedPrimary": Object {
        "&:not(:last-child)": Object {
          "&.Mui-disabled": Object {
            "borderColor": "#0000003D",
          },
          "borderColor": "#EF8426",
        },
      },
      "groupedContainedSecondary": Object {
        "&:not(:last-child)": Object {
          "&.Mui-disabled": Object {
            "borderColor": "#0000003D",
          },
          "borderColor": "#FF5808",
        },
      },
      "groupedTextHorizontal": Object {
        "&:not(:last-child)": Object {
          "&.Mui-disabled": Object {
            "borderColor": "#0000003D",
          },
          "borderRightColor": "#0000001F",
        },
      },
      "groupedTextPrimary": Object {
        "&:not(:last-child).Mui-disabled": Object {
          "borderColor": "#0000003D",
        },
      },
      "groupedTextSecondary": Object {
        "&:not(:last-child).Mui-disabled": Object {
          "borderColor": "#0000003D",
        },
      },
    },
    "MuiChip": Object {
      "clickableColorPrimary": Object {
        "&:hover": Object {
          "backgroundColor": "#EF8426",
        },
      },
      "clickableColorSecondary": Object {
        "&:hover": Object {
          "backgroundColor": "#EF8426",
        },
      },
      "deleteIcon": Object {
        "&:hover": Object {
          "color": "#000000",
        },
        "color": "rgba(0, 0, 0, 0.8)",
        "fontSize": "24px",
        "height": "24px",
        "width": "24px",
      },
      "deleteIconColorPrimary": Object {
        "&:hover": Object {
          "color": "#FFFFFF",
        },
        "color": "rgba(255, 255, 255, 0.8)",
      },
      "deleteIconColorSecondary": Object {
        "&:hover": Object {
          "color": "#FFFFFF",
        },
        "color": "rgba(255, 255, 255, 0.8)",
      },
      "deleteIconSmall": Object {
        "fontSize": "16px",
        "height": "16px",
        "width": "16px",
      },
      "icon": Object {
        "fontSize": "24px",
      },
      "iconSmall": Object {
        "fontSize": "16px",
        "height": "16px",
        "marginTop": "1px",
        "width": "16px",
      },
      "outlined": Object {
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "color": "#000000",
      },
      "outlinedPrimary": Object {
        "color": "#000000",
      },
      "outlinedSecondary": Object {
        "color": "#000000",
      },
      "root": Object {
        "& $avatar": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.24)",
          "color": "#333333",
        },
        "&$disabled": Object {
          "opacity": 0.4,
        },
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "color": "#333333",
        "fontSize": "0.875rem",
      },
    },
    "MuiExpansionPanel": Object {
      "root": Object {
        "&$disabled": Object {
          "backgroundColor": "#FFFFFF",
          "opacity": 0.32,
        },
        "&$expanded": Object {
          "borderColor": "#F4AB34",
        },
        "border": "transparent 1px solid",
      },
    },
    "MuiExpansionPanelSummary": Object {
      "root": Object {
        "&$disabled": Object {
          "opacity": 1,
        },
        "alignItems": "flex-start",
        "marginTop": "8px",
      },
    },
    "MuiFormLabel": Object {
      "formControl": Object {
        "marginTop": "-4px",
        "transform": "none",
      },
      "root": Object {
        "&.Mui-error": Object {
          "color": "#E74627",
        },
        "&.Mui-focused": Object {
          "color": "#777777",
          "fontSize": "0.875rem",
        },
        "fontSize": "0.875rem",
      },
    },
    "MuiIconButton": Object {
      "root": Object {
        "padding": "8px",
      },
      "sizeSmall": Object {
        "padding": "4px",
      },
    },
    "MuiInputBase": Object {
      "input": Object {
        "color": "#333333",
      },
      "root": Object {
        "&.Mui-focused": Object {
          "boxShadow": "#F4AB34 0 0 0 2px",
        },
        "&.MuiInput-underline.Mui-disabled:before": Object {
          "borderBottomStyle": "none",
        },
        "&.MuiInput-underline:after": Object {
          "borderBottom": "none",
        },
        "&.MuiInput-underline:before": Object {
          "borderBottom": "none",
        },
        "&.MuiInput-underline:hover:not(.Mui-disabled):before": Object {
          "borderBottom": "none",
        },
        "&:hover": Object {
          "boxShadow": "#777777 0 0 0 1px",
        },
        "&:hover.Mui-disabled": Object {
          "boxShadow": "#BBBBBB 0 0 0 1px",
        },
        "borderRadius": "4px",
        "boxShadow": "#BBBBBB 0 0 0 1px",
        "color": "#333333",
        "padding": "12px",
      },
    },
    "MuiInputLabel": Object {
      "formControl": Object {
        "marginTop": "-4px",
        "transform": "none",
      },
      "shrink": Object {
        "marginTop": "-4px",
        "transform": "none",
      },
    },
    "MuiListItem": Object {
      "button": Object {
        "&:hover": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.04)",
        },
      },
      "gutters": Object {
        "&:focus": Object {
          "outline": "none",
        },
      },
      "root": Object {
        "&$selected": Object {
          "&:focus": Object {
            "outline": "none",
          },
          "&:hover": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.04)",
          },
          "backgroundColor": "rgba(255, 107, 11, 0.16)",
        },
        "&.Mui-disabled": Object {
          "opacity": 0.32,
        },
        "&:hover": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.04)",
        },
        "cursor": "pointer",
      },
    },
    "MuiSelect": Object {
      "select": Object {
        "&.Mui-disabled": Object {
          "color": "#BBBBBB",
        },
        "&:focus": Object {
          "backgroundColor": "transparent",
        },
      },
    },
    "MuiTab": Object {
      "root": Object {
        "&$disabled": Object {
          "opacity": 0.24,
        },
      },
      "textColorInherit": Object {
        "&$disabled": Object {
          "opacity": 0.24,
        },
      },
      "wrapper": Object {
        "& > *:first-child": Object {
          "marginBottom": "2px !important",
        },
      },
    },
    "MuiTextField": Object {
      "& .MuiInputLabel": Object {
        "marginTop": "20px",
      },
    },
    "MuiToolbar": Object {
      "dense": Object {
        "@media (min-width: 600px)": Object {
          "minHeight": "56px",
        },
        "minHeight": "56px",
      },
      "gutters": Object {
        "@media (min-width: 600px)": Object {
          "padding": "0 16px",
        },
        "padding": "0 16px",
      },
      "regular": Object {
        "@media (min-width: 600px)": Object {
          "minHeight": "56px",
        },
        "minHeight": "56px",
      },
      "root": Object {
        "@media (min-width: 600px)": Object {
          "padding": "0 16px",
        },
        "padding": "0 16px",
      },
    },
  },
  "palette": Object {
    "action": Object {
      "disabled": "rgba(51, 51, 51, 0.24)",
    },
    "background": Object {
      "default": "#FAFAFA",
      "defaultContrastText": "#333333",
      "paper": "#FFFFFF",
      "paperContrastText": "#333333",
    },
    "complementary": Object {
      "highlight": "#000000",
      "link": "#227BBD",
      "linkContrastText": "#FFFFFF",
      "warning": "#FCC433",
      "warningContrastText": "#333333",
    },
    "error": Object {
      "contrastText": "#FFFFFF",
      "main": "#E74627",
    },
    "primary": Object {
      "contrastText": "#333333",
      "dark": "#EF8426",
      "light": "#FEFDE8",
      "main": "#F4AB34",
    },
    "secondary": Object {
      "contrastText": "#000000",
      "dark": "#FF5808",
      "light": "#FFF8E1",
      "main": "#FF6B0B",
    },
    "success": Object {
      "contrastText": "#FFFFFF",
      "main": "#569A32",
    },
    "text": Object {
      "disabled": "#BBBBBB",
      "hint": "#BBBBBB",
      "primary": "#333333",
      "secondary": "#777777",
    },
  },
  "shadows": Array [
    "none",
    "0 1px 1px 0 rgba(0, 0, 0, 0.14)",
    "0 2px 2px 0 rgba(0, 0, 0, 0.14)",
    "0 3px 4px 0 rgba(0, 0, 0, 0.14)",
    "0 4px 5px 0 rgba(0, 0, 0, 0.14)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
    "0 6px 10px 0 rgba(0, 0, 0, 0.14)",
    "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
    "0 8px 10px 0 rgba(0, 0, 0, 0.14)",
    "0 9px 12px 0 rgba(0, 0, 0, 0.14)",
    "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
    "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
    "0 12px 17px 0 rgba(0, 0, 0, 0.14)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
    "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
    "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
    "0 16px 24px 0 rgba(0, 0, 0, 0.14)",
    "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
    "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
    "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
    "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
    "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
    "0 24px 38px 0 rgba(0, 0, 0, 0.14)",
  ],
  "shape": Object {
    "badgeBorderRadius": 100,
    "borderRadius": 4,
  },
  "sizes": Object {
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "medium": 48,
    "mediumx": 56,
    "micro": 4,
    "none": 0,
    "semi": 32,
    "semix": 40,
    "small": 16,
    "standard": 24,
    "tiny": 8,
    "veryhuge": 256,
  },
  "spacing": 8,
  "typography": Object {
    "body1": Object {
      "fontSize": "1rem",
      "fontWeight": 400,
    },
    "body2": Object {
      "fontSize": "0.875rem",
      "fontWeight": 400,
    },
    "button": Object {
      "fontSize": "0.875rem",
      "fontWeight": 600,
    },
    "caption": Object {
      "fontSize": "0.75rem",
      "fontWeight": 400,
    },
    "fontFamily": "\\"Roboto\\", sans-serif",
    "fontFamilyBrand1": "\\"Helvetica Now Display\\"",
    "fontFamilyBrand2": "\\"Helvetica Now Text\\"",
    "fontFamilyBrand3": undefined,
    "fontWeightBold": 700,
    "fontWeightLight": 300,
    "fontWeightMedium": 500,
    "fontWeightRegular": 400,
    "h1": Object {
      "fontSize": "6rem",
      "fontWeight": 300,
    },
    "h2": Object {
      "fontSize": "3.75rem",
      "fontWeight": 300,
    },
    "h3": Object {
      "fontSize": "3rem",
      "fontWeight": 400,
    },
    "h4": Object {
      "fontSize": "2.125rem",
      "fontWeight": 400,
    },
    "h5": Object {
      "fontSize": "1.5rem",
      "fontWeight": 400,
    },
    "h6": Object {
      "fontSize": "1.25rem",
      "fontWeight": 600,
    },
    "overline": Object {
      "fontSize": "0.75rem",
      "fontWeight": 400,
    },
    "subtitle1": Object {
      "fontSize": "1rem",
      "fontWeight": 400,
    },
    "subtitle2": Object {
      "fontSize": "0.875rem",
      "fontWeight": 500,
    },
  },
}
`;
